File: src/main/java/com/kaankaplan/booke/business/abstracts/BookService.java
package com.kaankaplan.booke.business.abstracts;

import com.kaankaplan.booke.core.util.results.DataResult;
import com.kaankaplan.booke.core.util.results.Result;
import com.kaankaplan.booke.dto.BookDto;
import com.kaankaplan.booke.modals.Book;
import com.kaankaplan.booke.modals.Review;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BookService {

    DataResult<Book> addBook(BookDto bookDto);

    DataResult<Book> addReviewToBook(String bookId, Review review);

    DataResult<Book> getBookById(String bookId);

    DataResult<List<Book>> searchBooksAutocomplete(String title);

    Result addImageForBook(String bookId, MultipartFile multipartFile);
}

File: src/main/booke-mobile-app/services/BookService.js
import Interceptor from "../utils/interceptors/Interceptor";

export default class BookService {
    apiUrl = "book/";
    axiosInstance;

    constructor() {
        this.axiosInstance = new Interceptor().getInstance();
    }

    getBookById(bookId) {
        return this.axiosInstance.get(this.apiUrl + "getBookById/" + bookId);
    }

    search(title) {
        console.log(this.apiUrl + "search/" + title)
        return this.axiosInstance.get(this.apiUrl + "search/" + title);
    }

    addReviewToBook(bookId, review) {
        return this.axiosInstance.post(this.apiUrl + "addReviewToBook/" + bookId, review);
    }
}